tree/treeplayer/test/dataframe/dataframe_callbacks.cxx

```diff
@@ -0,0 +1,72 @@
+#include "ROOT/TDataFrame.hxx"
+#include "TRandom.h"
+#include "gtest/gtest.h"
+using namespace ROOT::Experimental;
+using namespace ROOT::Experimental::TDF;
+using namespace ROOT::Detail::TDF;
+
+// fixture that provides a TDF with no data-source and a single column "x" containing normal-distributed doubles
+class TDFCallbacks : public ::testing::Test {
+protected:
+   const ULong64_t nEvents = 8ull; // must be initialized before fLoopManager
+
+private:
+   TDataFrame fLoopManager;
+   TInterface<TLoopManager> DefineRandomCol()
+   {
+      TRandom r;
+      return fLoopManager.Define("x", [r]() mutable { return r.Gaus(); });
+   }
+
+protected:
+   TDFCallbacks() : fLoopManager(nEvents), tdf(DefineRandomCol()) {}
+   TInterface<TLoopManager> tdf;
+};
+
+TEST_F(TDFCallbacks, Histo1DWithFillTOHelper)
+{
+   auto h = tdf.Histo1D<double>({"", "", 128, -2., 2.}, "x");
+   using value_t = typename decltype(h)::Value_t;
+   ULong64_t i = 0ull;
+   ULong64_t everyN = 1ull;
+   h.OnPartialResult(everyN, [&i, &everyN](value_t &h_) {
```

Comment:
Should this be extended to also test OnPartialResultSlot. ?

Reply:
yes, and all the different actions! I will lift the `do not merge` once support for jitted actions and all the necessary tests are in.